### PR TITLE
WS2-1847 : Check for/add missing horizontal cards

### PR DESCRIPTION
**Pull request by juanmitriatti**
### Description

We have some horizontal cards, such as the horizontal story card which aren’t represented in WS2. For this ticket:

take stock of what’s missing

implement the missing items - we can discuss post-scrum the best approach for implementing the components. 

To be added as individual cards like the Card Image and Content component (instead of arrangements which force columns that don’t make sense for  horizontal cards). 

Solution :
Creating new functionality to allow horizontal display for some Card arrangement options

**QA Steps:**

1. Add a new “Card Arrangement” block

2. Ensure the presence of the recently added feature "Display orientation."

. If you choose “Add Card Group Ranking” or “Add Image based cards” then “Display orientation”  must be hidden. That option won’t be displayed for those two options.

.  If you choose another Card type and  then “Display orientation = Horizontal”  the  “Columns to Display” select have only “Two Columns” option.

### Links

- [JIRA ticket](https://asudev.jira.com/browse/WS2-1847)

### Checklist

- [ ] Design updates match [Web Standards](https://xd.adobe.com/view/56f6cb78-9af5-4b12-b4ce-ef319f71113f-03a5/) and [Unity Design System](https://unity.web.asu.edu)
- [ ] Solution is documented on the Jira ticket
- [ ] QA steps to verify have been included on the Jira ticket
- [ ] No new PHP or JS errors
- [ ] No accessibility issues are introduced with this update
- [ ] Added/updated README.md files, if relevant

### Verified in browsers 

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Screenshots

Note: Sections that are not applicable for this issue can be removed.

### CODE REVIEW

**Comment by mlsamuelson** [comment]
[Visit Site](https://pr-223-webspark-ci.pantheonsite.io/)

Created multidev environment [pr-223](https://dashboard.pantheon.io/sites/aec2f75f-35eb-41f8-abe7-95b292415259#pr-223) for webspark-ci.

File: web/modules/webspark/webspark_blocks/js/card-arrangement.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      const elementBasedCard = document.querySelector('[name*="[field_loading]"]');
+
+      // Check if the element exists to determine display orientation
+      if (elementRanking || elementBasedCard) {
+        // Hide display orientation
+        _showDisplayOrientation('none');
+      } else {
+        // Show display orientation
+        _showDisplayOrientation();
+      }
```
Comment (mlsamuelson) [inline web/modules/webspark/webspark_blocks/js/card-arrangement.js:22]: This seems to be working fine, but I'm wondering if you considered using the [Drupal Conditional Form States API](https://www.drupal.org/docs/drupal-apis/form-api/conditional-form-fields) to manage the form field logic? I believe everything this script does could be implemented using that API.

Example of this being used in the `asu_degree_rfi` module is at https://github.com/ASUWebPlatforms/webspark-ci/blob/master/web/modules/webspark/asu_degree_rfi/asu_degree_rfi.module#L90-L110

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
The functionality tests well, and handling the default value for the "Display orientation" field so that we don't regress existing instances is good, including requiring the user to explicitly set the value the next time they edit the block.

I'd like to request we change the label on that field to "Card orientation" instead of "Display orientation" to avoid confusion because just a few fields away we have the "View mode" field.

There are a few other changes requested:
* I've added a comment where we need to remove some of the UUID data from one of the .yml files.
* I asked in another comment if there is specific reason for not using Drupal's conditional field functionality instead of writing our own.

File: web/modules/webspark/webspark_blocks/js/card-arrangement.js (hunk 1 above)
Comment (mlsamuelson) [reply web/modules/webspark/webspark_blocks/js/card-arrangement.js:22]: This comment is referring to the whole card-arrangement.js file.

**Comment by mlsamuelson** [comment]
On further thought, since this is a new field, what do you think about having the value default to vertical, since that should be all the exists to start with.  That will avoid people having to deal with form validation errors every time they update an existing card arrangement. The current setup is okay. But if it doesn't take much more work, and doesn't introduce the possibility of regressions, let's default to vertical.

**Comment by juanmitriatti** [comment]
> On further thought, since this is a new field, what do you think about having the value default to vertical, since that should be all the exists to start with. That will avoid people having to deal with form validation errors every time they update an existing card arrangement. The current setup is okay. But if it doesn't take much more work, and doesn't introduce the possibility of regressions, let's default to vertical.



Hi @mlsamuelson 
As for this suggestion, when you create a new Card Arrangement, the default value for Card Orientation is vertical. However, the one that are already created don't have any value assigned. Is that what you suggested ? I believe, it'll require a hook update for all those already created blocks.

File: web/modules/webspark/webspark_blocks/config/install/core.entity_view_display.block_content.card_arrangement.event_card.yml [hunk 2]
```diff
@@ -18,8 +19,10 @@ dependencies:
     - layout_builder
 third_party_settings:
   layout_builder:
-    allow_custom: false
     enabled: false
+    allow_custom: false
+_core:
+  default_config_hash: X5e0iH8NVMibexlUWXX_IU4kEcRDpLKqVNYbB3-Nu6U
```
[the remark this answers is not in the document]
Comment (juanmitriatti) [reply web/modules/webspark/webspark_blocks/config/install/core.entity_view_display.block_content.card_arrangement.event_card.yml]: This is done @mlsamuelson 
Thanks!

File: web/modules/webspark/webspark_blocks/js/card-arrangement.js (hunk 1 above)
Comment (mlsamuelson) [reply web/modules/webspark/webspark_blocks/js/card-arrangement.js:22]: Thank you for checking! I just wanted to make sure we weren't "reinventing the wheel" but sounds like our custom code is required. :)

**Comment by mlsamuelson** [comment]
Let's leave the no-default value for existing cards as-is.

**Review by mlsamuelson — APPROVED** [review]
Approved.

**Pull request merged**